peas/sensors.py

```diff
@@ -151,46 +90,35 @@ def capture(self, use_mongo=True, send_message=True):
             sensor_data (dict):     Dictionary of sensors keyed by sensor name.
         """
 
+        # Read from all the readers; we send messages with sensor data immediately, but accumulate
+        # data from all sensors before storing in the db.
+        # Note that there is no guarantee that these are the LATEST reports emitted by the sensors,
+        # as the 
         sensor_data = dict()
-
-        # Read from all the readers
         for sensor_name, reader_info in self.serial_readers.items():
             reader = reader_info['reader']
 
-            # Get the values
-            self.logger.debug('Reading next serial value')
+            # Get the values before attempting to re
```

Comment:
Seems like your comments are having troubles finishing themselves today. :)

Reply:
My thoughts are leaping all over the ...